lib/atlas/dataset/derived_dataset.rb

```diff
@@ -0,0 +1,63 @@
+module Atlas
+  class Dataset::DerivedDataset < Dataset
+    GRAPH_FILENAME = 'graph.yml'.freeze
+
+    attribute :base_dataset, String
+    attribute :scaling,      Preset::Scaling
+    attribute :init,         Hash[Symbol => Float]
+
+    validates :scaling, presence: true
+
+    validate :base_dataset_exists
```

Comment:
Could these methods be named "validate_something" for consistency with existing validation methods in Edge, Input, Node, etc. It helps to differentiate validation methods from "normal" methods.